docs/form-datasets.rst

```diff
@@ -1,32 +1,47 @@
+* data comes from another system. Using external datasets generally requires fewer steps than adding the data to a form definition.
+* data needs to change over time. External datasets can be updated without modifying the form definition.
+* data is reused between forms. It may be easier to attach the same data file to multiple forms instead of copying the data into all the form definitions. Entity Lists can be shared between forms.
+* the same forms are used in different contexts. For example, the exact same form definition could be used in multiple countries with different data files or Entity Lists listing regions, local products, etc.
 
-All of the techniques described below apply to :doc:`Entity Lists <central-entities>`. Entity Lists can be thought of as CSVs that are managed by ODK Central so that data collected by one form submission can be automatically accessible in other forms in the same project.
+The more dynamic the data is, the more likely you are to benefit from Entity Lists over attached files.
 
 .. note::
 
-  Most mobile devices released in 2019 or later can handle lists of 50,000 or more without slowing down. If you experience Collect slowing down, please share the size of the dataset, the device you are using, and any expressions that reference the dataset on `the community forum <https://forum.getodk.org/c/support/6>`_ or to support@getodk.org.
+  Most mobile devices released in 2019 or later can handle lists of 50,000 or more elements without slowing down. If you experience Collect slowing down, please share the size of the dataset, the device you are using, and any expressions that reference the dataset on `the community forum <https://forum.getodk.org/c/support/6>`_ or to support@getodk.org.
 
 .. _selects-from-csv:
 
 Building selects from CSV files
 ---------------------------------
 
-CSV files can be used as datasets for select questions using ``select_one_from_file`` or ``select_multiple_from_file``. CSV files used this way must have ``name`` and ``label`` columns unless you use the ``parameters`` column to :ref:`customize the columns used <customizing-label-and-value>`. For each row in the dataset, the text in the value column (defaults to ``name``) will be used as the value saved when that choice is selected and the text in the label column (defaults to ``label``) will be used to display the choice. For select multiples, ``name`` must not contain spaces.
+CSV files can be used as choice lists for select questions using ``select_one_from_file`` or ``select_multiple_from_file``. CSV files used this way must have ``name`` and ``label`` columns or you use the ``parameters`` column to :ref:`customize the columns used <customizing-label-and-value>`. For each row in the dataset, the text in the value column (defaults to ``name``) will be used as the value saved when that choice is selected and the text in the label column (defaults to ``label``) will be used to display the choice.
+
+.. note::
```

Comment:
The "must" suggests to me that it should be a warning not a note, but fine if you disagree.